controller/resource_roles.go

```diff
@@ -69,6 +82,62 @@ func (c *ResourceRolesController) ListAssignedByRoleName(ctx *app.ListAssignedBy
 	})
 }
 
+// AssignRole assigns a specific role for a resource, to one or more identities.
+func (c *ResourceRolesController) AssignRole(ctx *app.AssignRoleResourceRolesContext) error {
+
+	currentUser, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError(err.Error()))
+	}
+	if currentUser == nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError("identity ID not found in token"))
+	}
+
+	// check if the current user token belongs to a user who has the necessary privileges
+	// for assigning roles to other users.
+	hasScope, err := c.permissionService.HasScope(ctx, *currentUser, ctx.ResourceID, ROLE_ASSIGNMENT_SCOPE)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+	if !hasScope {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("user is not authorized to assign roles"))
+	}
+
+	// In a batch assignment of roles, all users need to be a part of the resource
+	// Only then futher role assignments would be allowed.
+	for _, identity := range ctx.Payload.Data {
```

Comment:
"Users need to be a part of resource" - this is confusing :)
What you are ding here is just checking if the user has any explicit role for the resource. I see the intention here for this endpoint. It's about managing existing roles rather then inviting to them. But what about inherited roles or roles mapped from the parent resources?
It looks like we need more sophisticated ListAssignmentsByIdentityAndResource method which will take care of it.

@sbryzak thoughts?

Reply:
@alexeykazakov very good question about how to handle inherited roles, I think we need to spend some time thinking about what impact this may have on the UI and invitation workflow.  As for the comment, I agree it's confusing. :)  I would rewrite it as something like this:

````
// In batch assignment of roles all selected users must have previously been assigned 
// privileges for the resource, otherwise the invitation workflow should be used instead
````